packages/document/docs/en/plugins/list/plugin-styled-components.mdx

```diff
@@ -0,0 +1,89 @@
+# Styled Components Plugin
+
+[styled-components](https://styled-components.com/) is a popular CSS-in-JS implementation library, which uses the new JavaScript feature [Tagged template](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates) to write component CSS styles.
+
+Rsbuild provides compile-time support for styled-components. For details, please refer to [rspack#styledComponents](https://www.rspack.dev/guide/loader.html#optionsrspackexperimentsstyledcomponents).
```

Comment:
We should tell user that this plugin is to `Improve the debugging experience and add server-side rendering support`, same as https://github.com/styled-components/babel-plugin-styled-components